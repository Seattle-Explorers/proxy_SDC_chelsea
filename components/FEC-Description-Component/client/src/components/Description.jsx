import React from 'react';
import PropTypes from 'prop-types';
import makeKey from '../../../util/makeKey';
import Title from './Title';
import Body from './Body';
import Amenity from './Amenity';
import Amenities from './Amenities';
import AmenitiesModal from './AmenitiesModal';
import SleepingArrangment from './SleepingArrangement';
import { DescriptionBox } from './styles/Description.style';

function makeArrangements(sleepingArrangements) {
  return sleepingArrangements.map(({ location, beds }) => (
    <SleepingArrangment key={makeKey('dsa')} location={location} beds={beds} />
  ));
}

function Description({ data, showModal }) {
  const {
    body, title, guests, bedrooms, beds, amenities, sleepingArrangements, user,
  } = data;
  const sleepingComponents = makeArrangements(sleepingArrangements);
  return (
    <DescriptionBox>
      <AmenitiesModal
        show={showModal}
        amenities={amenities}
      />
      <Title title={title} guests={guests} bedrooms={bedrooms} beds={beds} user={user} />
      <Body>{body}</Body>
      {sleepingComponents}
      <Amenities amenities={amenities} />
    </DescriptionBox>
  );
}

Description.propTypes = {
  showModal: PropTypes.bool,
  data: PropTypes.shape({
    body: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    guests: PropTypes.number.isRequired,
    bedrooms: PropTypes.number.isRequired,
    beds: PropTypes.number.isRequired,
    amenities: PropTypes.arrayOf(PropTypes.shape(Amenity.propTypes)),
    sleepingArrangements: PropTypes.arrayOf(PropTypes.shape(SleepingArrangment.propTypes)),
    user: PropTypes.shape(Title.propTypes.user),
  }).isRequired,
};

Description.defaultProps = {
  showModal: false,
};

export default Description;
