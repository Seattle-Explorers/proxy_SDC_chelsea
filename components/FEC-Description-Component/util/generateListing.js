// const fs = require('fs');
// const fsp = require('fs').promises;
const path = require('path');
// const axios = require('axios');
const _ = require('lodash');
const faker = require('faker');
const { bedStrings, amenityStrings, titleStrings } = require('./seedStrings.js');

// call fn(time) a random number of times from max to min (inclusive)
// assumes max and min are >= 1
function randomIterations(min, max, fn) {
  if (max < 1 || min < 1) {
    throw (new Error('max and/or min must be >= 1 in randomIterations'));
  }
  // range(1, 1) makes an empty list
  // so I add one to max/min ensure there's always at least element
  return _.range(1, _.random(min + 1, max + 1)).map(fn);
}

// returns a promise that will resolve to a user object
async function generateUser(userId) {
  const name = `${faker.name.firstName()} ${faker.name.lastName()}`;
  const imageURL = 'http://lorempixel.com/56/56/';
  // `https://github.com/Team-Danger/FEC-Description-Component/util/profileImages/${userId}.jpg`;
  // const image = await axios({
  //   method: 'get',
  //   url: 'http://lorempixel.com/56/56/',
  //   responseType: 'stream',

  // });
  // fsp.stat(imagePath)
  //   .catch((err) => {
  //     if (err.code === 'ENOENT') {
  //       return fsp.mkdir(imagePath);
  //     }
  //     throw err;
  //   })
  //   .then(() => {
  //     image.data.pipe(fs.createWriteStream(path.join(imagePath, `${userId}.jpg`)));
  //   });
  return {
    userId,
    name,
    image: imageURL,
  };
}

function generateTitle() {
  const adjective = _.sample(titleStrings.adjective);
  const place = _.sample(titleStrings.place);
  const location = _.sample(titleStrings.location);
  return `${adjective} ${place} in ${location}`;
}

function generateArrangements() {
  let hasCommonArea = false;
  const arrangements = [];
  randomIterations(1, _.random(2, 10), (index) => {
    let location;
    // every iteration we have a 1 in 3 chance of adding a common area
    if (!_.random(0, 3) && !hasCommonArea) {
      hasCommonArea = true;
      location = 'Common Space';
    } else if (hasCommonArea) {
      // if there is a common area it caused us to skip an index
      // so we have to do this to make the bedrooms count up right
      location = `Bedroom${index - 1}`;
    } else {
      // there's no common area so the index matches the bedroom number
      location = `Bedroom${index}`;
    }
    const beds = [];
    randomIterations(1, 5, () => {
      beds.push({ amount: _.random(1, 5), type: _.sample(bedStrings) });
    });
    arrangements.push({
      location,
      beds,
    });
  });
  return arrangements;
}

// returns a promise resolving to a list of 5 to 20(ish) non repeating amenities
function generateAmenities() {
  const amenities = [];
  const unusedAmenities = JSON.parse(JSON.stringify(amenityStrings));
  randomIterations(5, 20, () => {
    const category = _.sample(unusedAmenities);
    // 50% chance of having a description
    const description = _.random(0, 1)
      ? faker.lorem.sentences(1)
      : '';
    const amenity = {
      type: category.type,
      amenity: _.sample(category.amenities),
      description,
    };
    if (amenity.amenity !== undefined) {
      amenities.push(amenity);
      category.amenities = _.without(category.amenities, amenity.amenity);
    }
  });
  return amenities;
}

// returns a promise that will resolve to a listing object
async function generateListing(
  listingId,
  userId,
  imageFolder = path.join(__dirname, 'profileImages'),
) {
  // ensure that there is always at least one bathroom
  // but make it possible to have 0 public or 0 private baths
  const publicBaths = _.random(0, 10);
  const privateBaths = _.random(publicBaths === 0 ? 1 : 0, 10);
  const arrangements = generateArrangements();
  let numBeds = 0;
  arrangements.forEach((bedroom) => {
    numBeds += bedroom.beds.length;
  });
  return {
    listingId,
    user: await generateUser(userId, imageFolder),
    title: generateTitle(),
    body: faker.lorem.paragraphs(5),
    guests: _.random(1, 10),
    bedrooms: arrangements.length,
    beds: numBeds,
    publicBaths,
    privateBaths,
    sleepingArrangements: arrangements,
    amenities: generateAmenities(),
  };
}
module.exports = generateListing;
