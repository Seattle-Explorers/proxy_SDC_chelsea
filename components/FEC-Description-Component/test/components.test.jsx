/**
 * @jest-environment jsdom
 */
import React from 'react';
import _ from 'lodash';
import faker from 'faker';
import { join } from 'path';
import { create } from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import generateListing from '../util/generateListing';
import Description from '../client/src/components/Description';
import 'babel-polyfill'; // this is needed to make async/await work
import 'jest-styled-components';

describe('component tests', () => {
  test('it should render a description component', async () => {
    jest.setTimeout(90000);
    // make sure we always generate the same listing object
    jest.unmock('lodash');
    _.random = (min) => min;
    _.sample = (collection) => collection[0];
    faker.seed(1337);
    const imagesPath = join(__dirname, 'testImages');
    const listing = await generateListing('001', '001', imagesPath);
    const component = (
      <Router>
        <Description data={listing} />
      </Router>
    );
    const tree = create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
