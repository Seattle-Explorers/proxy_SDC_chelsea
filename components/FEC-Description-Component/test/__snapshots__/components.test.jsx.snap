// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component tests it should render a description component 1`] = `
.c9 {
  padding-top: 32px;
  padding-bottom: 48px;
  border-bottom: 1px solid #ddd;
}

.c0 {
  font-family: -apple-system,BlinkMacSystemFont,Roboto,"Helvetica Neue",sans-serif !important;
  font-weight: 400 !important;
  font-size: 16px !important;
  line-height: 24px !important;
  width: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  color: rgb(34,34,34) !important;
  font-weight: 600 !important;
  font-size: 16px !important;
  line-height: 20px !important;
  margin-bottom: 8px !important;
}

.c13 {
  color: #222222 !important;
  display: block;
  font-weight: 400 !important;
  font-size: 16px !important;
  line-height: 20px !important;
}

.c14 {
  border-radius: 50%;
}

.c15 {
  font-size: inherit !important;
  font-family: inherit !important;
  font-style: inherit !important;
  line-height: inherit !important;
  cursor: pointer !important;
  -webkit-user-select: auto !important;
  -moz-user-select: auto !important;
  -ms-user-select: auto !important;
  user-select: auto !important;
  color: rgb(34,34,34) !important;
  font-weight: 600 !important;
  background: transparent !important;
  border-width: 0px !important;
  border-style: initial !important;
  border-color: initial !important;
  border-image: initial !important;
  margin: 0px !important;
  padding: 0px !important;
  -webkit-text-decoration: underline !important;
  text-decoration: underline !important;
  border-radius: 4px !important;
  outline: none !important;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 50%;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-bottom: 16px;
}

.c23 {
  margin-right: 16px !important;
  margin-left: 0px !important;
}

.c24 {
  margin-top: 24px;
  cursor: pointer !important;
  display: inline-block !important;
  margin: 0px !important;
  text-align: center !important;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  width: auto !important;
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
  font-size: 16px !important;
  line-height: 20px !important;
  font-weight: 600 !important;
  border-radius: 8px !important;
  border-width: 1px !important;
  border-style: solid !important;
  outline: none !important;
  padding-top: 13px !important;
  padding-bottom: 13px !important;
  padding-left: 23px !important;
  padding-right: 23px !important;
  -webkit-transition: box-shadow 0.2s ease,-ms-transform 0.1s ease,-webkit-transform 0.1s ease,-webkit-transform 0.1s ease !important;
  -webkit-transition: box-shadow 0.2s ease,-ms-transform 0.1s ease,-webkit-transform 0.1s ease,transform 0.1s ease !important;
  transition: box-shadow 0.2s ease,-ms-transform 0.1s ease,-webkit-transform 0.1s ease,transform 0.1s ease !important;
  border-color: #222222 !important;
  background: #FFFFFF !important;
  color: #222222 !important;
}

.c1 {
  display: none;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0,0,0,0.6);
}

.c2 {
  border-radius: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 780px;
  position: fixed;
  height: 900px;
  background: white;
  width: 100%;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 24px;
  overflow-y: scroll;
  margin-bottom: 24px;
}

.c6 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100% !important;
  margin-top: 24px !important;
  margin-bottom: 24px !important;
  border-bottom: 1px solid #ddd;
}

.c7 {
  font-size: 14px;
  line-height: 1.43;
}

.c8 {
  color: rgb(113,113,113) !important;
  font-weight: 400 !important;
  font-size: 14px !important;
  line-height: 18px !important;
  margin-top: 4px !important;
}

.c3 {
  -webkit-box-flex: 0 !important;
  -webkit-flex-grow: 0 !important;
  -ms-flex-positive: 0 !important;
  flex-grow: 0 !important;
  -webkit-flex-shrink: 0 !important;
  -ms-flex-negative: 0 !important;
  flex-shrink: 0 !important;
  -webkit-flex-basis: 72px !important;
  -ms-flex-preferred-size: 72px !important;
  flex-basis: 72px !important;
  position: -webkit-sticky !important;
  position: sticky !important;
  text-align: left !important;
  top: 0px !important;
  z-index: -1 !important;
}

.c4 {
  top: 24px;
  left: 24px;
  position: absolute;
  display: inline-block !important;
  color: rgb(34,34,34) !important;
  cursor: pointer !important;
  position: relative !important;
  border-radius: 50% !important;
  border-width: initial !important;
  border-style: none !important;
  border-color: initial !important;
  border-image: initial !important;
  outline: 0px !important;
  margin: 0px !important;
  padding: 0px !important;
  background: transparent !important;
  -webkit-transition: -ms-transform 0.25s ease 0s,-webkit-transform 0.25s ease 0s,-webkit-transform 0.25s ease 0s !important;
  -webkit-transition: -ms-transform 0.25s ease 0s,-webkit-transform 0.25s ease 0s,transform 0.25s ease 0s !important;
  transition: -ms-transform 0.25s ease 0s,-webkit-transform 0.25s ease 0s,transform 0.25s ease 0s !important;
}

.c16 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #ddd;
  border-radius: 12px;
}

.c17 {
  margin-bottom: 8px;
}

.c19 {
  color: rgb(34,34,34) !important;
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,"Helvetica Neue",sans-serif !important;
  font-weight: 400 !important;
  font-size: 14px !important;
  line-height: 20px !important;
}

.c18 {
  color: rgb(34,34,34) !important;
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,"Helvetica Neue",sans-serif !important;
  font-weight: 600 !important;
  font-size: 16px !important;
  line-height: 20px !important;
  margin-bottom: 8px;
}

<div
  className="c0"
>
  <section
    className="c1"
    onClick={[Function]}
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <div
        className="c3"
      >
        <button
          className="c4"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="c5"
      >
        <h2>
          Amenities
        </h2>
        <div>
          <h3>
            Basic
          </h3>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              Essentials
            </div>
            <div
              className="c8"
            >
              
            </div>
          </div>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              Wifi
            </div>
            <div
              className="c8"
            >
              
            </div>
          </div>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              TV
            </div>
            <div
              className="c8"
            >
              
            </div>
          </div>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              Heat
            </div>
            <div
              className="c8"
            >
              
            </div>
          </div>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              Air Conditioning
            </div>
            <div
              className="c8"
            >
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="c9"
  >
    <div
      className="c10"
    >
      <div
        className="c11"
      >
        <div
          className="c12"
        >
          Cozy apartment in Portland hosted by Devyn Macejkovic
        </div>
        <div
          className="c13"
        >
          1 guest · 1 bedroom · 1 bed
        </div>
      </div>
      <img
        alt="user"
        className="c14"
        src="http://lorempixel.com/56/56/"
      />
    </div>
  </div>
  <div
    className="c9"
  >
    <span
      width={0}
    >
      <span />
      <span>
        Minima perspiciatis praesentium aperiam. Ut occaecati deserunt voluptatem suscipit. Enim minus nemo fugit et iure explicabo et commodi consequuntur.
 
Nesciunt dignissimos eos temporibus temporibus autem consequatur ea laudantium esse. Qui occaecati omnis quis amet voluptatum. Et libero ea et odio. Qui velit qui velit. Provident non fugiat dolorum amet a a aut nihil vel.
 
Excepturi eligendi perferendis quo minus et asperiores nam. Blanditiis consequuntur esse autem. Eligendi aut alias cum consequatur pariatur inventore omnis. Sapiente aliquid illo cumque cupiditate quos cum aut placeat. Ut est ut error quo repellat id aliquam eos.
 
Sunt est itaque ad aut. Quos quis laboriosam molestias quisquam ratione aut est omnis necessitatibus. Ut consectetur voluptas nam praesentium voluptates libero neque.
 
Temporibus quia officiis cumque eos ut nemo occaecati voluptatibus dolor. Totam praesentium ducimus tempore qui rerum autem atque. Esse est inventore autem eligendi aliquid. Voluptates quis quo. Eveniet deleniti quam sed totam nam et.
      </span>
      <span
        style={
          Object {
            "left": 0,
            "position": "fixed",
            "top": 0,
            "visibility": "hidden",
          }
        }
      >
        <span>
          ...
          <button
            className="c15"
            onClick={[Function]}
            type="button"
          >
            read more
          </button>
        </span>
      </span>
    </span>
  </div>
  <div
    className="c9"
  >
    <div
      className="c16"
    >
      <div
        className="c17"
      >
        <SvgrURL />
      </div>
      <div
        className="c18"
      >
        Common Space
      </div>
      <div
        className="c19"
      >
        1 Double
      </div>
    </div>
  </div>
  <div
    className="c9"
  >
    <div
      className="c20"
    >
      <div
        className="c21"
      >
        <div
          className="c22"
        >
          <div
            className="c23"
          >
            <SvgrURL />
          </div>
          <div>
            Essentials
          </div>
        </div>
        <div
          className="c22"
        >
          <div
            className="c23"
          >
            <SvgrURL />
          </div>
          <div>
            Wifi
          </div>
        </div>
        <div
          className="c22"
        >
          <div
            className="c23"
          >
            <SvgrURL />
          </div>
          <div>
            TV
          </div>
        </div>
        <div
          className="c22"
        >
          <div
            className="c23"
          >
            <SvgrURL />
          </div>
          <div>
            Heat
          </div>
        </div>
        <div
          className="c22"
        >
          <div
            className="c23"
          >
            <SvgrURL />
          </div>
          <div>
            Air Conditioning
          </div>
        </div>
      </div>
      <div
        className="c21"
      />
    </div>
    <a
      className="c24"
      href="/amenities"
      onClick={[Function]}
    >
      All Amenities
    </a>
  </div>
</div>
`;
